### ISSUE TITLE
Winter Wyvern - Winter's Curse kill undetected

### ISSUE THREAD

**Issue by mvroe** [ValveSoftware/Dota2-Gameplay#7600]
#### Your system information

* Have you checked for system updates?: Yes
* Are you using the latest stable video driver available for your system? Yes
* Have you verified the game files?: No

#### Please describe your issue in as much detail as possible:
After killing Lycan (5:40.433) no kill was awarded to Winter Wyvern. No death / kill message was shown in the game client. The death counter was increased by 1 for Lycan, but no kill or assist was awarded to Winter Wyvern. No bounty was awarded.

Probable cause could be the flying projectile from the Dire's siege creep dealing the killing blow after Winter's Curse debuff has expired.

Match Id: 7064184425
Timestamp 5:35-5:41 

![Screenshot 2023-03-17 114621](https://user-images.githubusercontent.com/9080594/225883825-5763cb95-e824-4992-8547-7951080308f7.png)

**Comment by shaileshgarg** [comment]
This is expected, Lycan didn't die under Winter's Curse debuff, so Winter didn't get the kill.

**Comment by mvroe** [comment]
@shaileshgarg That is incorrect. In case of solo hero assisted with creeps, the hero is always awarded the kill for the enemy hero. To emphazize, it was the Dire Siege Creep that did the killing blow to the Dire Lycan.

**Comment by TheUnholyWrath** [comment]
#1389. Its the same mechanic, neutral creeps getting the kill makes this happen. Allies or allied controlled units results in a deny.

**Comment by mvroe** [comment]
@TheUnholyWrath Thank you for linking the related issues. Apart from the ruleset wether the kill should be awarded or not, at least there should've been a chat message stating that Lycan was denied by Siege Creep, like with other heroes denies.

**Comment by geomag43** [comment]
Another instance https://github.com/ValveSoftware/Dota2-Gameplay/issues/26778
